packages/@aws-cdk/aws-iam/lib/oidc-provider.ts

```diff
@@ -135,24 +140,31 @@ export class OpenIdConnectProvider extends Resource implements IOpenIdConnectPro
   public constructor(scope: Construct, id: string, props: OpenIdConnectProviderProps) {
     super(scope, id);
 
+    const provider = this.getOrCreateProvider();
     const resource = new CustomResource(this, 'Resource', {
       resourceType: RESOURCE_TYPE,
-      serviceToken: this.getOrCreateProvider(),
+      serviceToken: provider.serviceToken,
       properties: {
         ClientIDList: props.clientIds,
         ThumbprintList: props.thumbprints,
         Url: props.url,
+
+        // code changes can cause thumbprint changes in case they weren't explicitly provided.
+        // add the code hash as a property so that CFN invokes the UPDATE handler in these cases,
+        // thus updating the thumbprint if necessary.
+        CodeHash: provider.codeHash,
```

Comment:
@rix0rrr Can I get your eyes on this? Basically because the provider dynamically fetches the thumbprint, its value can change due to code changes, even when the properties of the resource haven't changed. So I found that this is required to make CFN trigger the UPDATE handler when the code changes. 

Am I missing something?